File: cybernovels/backup/pandoc_cli/doconvrt.jl
#!/usr/bin/env julia

using ArgParse

function main()
    s = ArgParseSettings(
        description="A universal document converter using pandoc.",
        epilog="""
        Supported formats include:
        - markdown (.md)
        - latex (.tex)
        - pdf (.pdf)
        - html (.html)
        - plain text (.txt)
        """
    )
    @add_arg_table! s begin
        "input"
            help = "Input file to convert"
            required = true
        "--output", "-o"
            help = "Output file"
            required = true
    end
    
    parsed_args = parse_args(ARGS, s)
    
    input_file = parsed_args["input"]
    output_file = parsed_args["output"]

    if isnothing(Sys.which("pandoc"))
        println(stderr, "Error: pandoc is not installed or not in the system's PATH.")
        exit(1)
    end
    
    cmd = `pandoc $(input_file) -o $(output_file) --pdf-engine=pdflatex -V fontsize=12pt -V documentclass=article -V geometry:margin=1in`

    try
        run(cmd)
        println("Successfully converted '$(input_file)' to '$(output_file)'")
    catch e
        println(stderr, "Error converting file: ", e)
        exit(1)
    end
end

# Only run main if this script is executed directly
if abspath(PROGRAM_FILE) == @__FILE__
    main()
end

File: cybernovels/backup/pandoc_cli/doconvrt.py
#!/usr/bin/env python3
import argparse
import subprocess
import sys
import shutil

def main():
    """
    A command-line tool to convert documents using pandoc.
    """
    parser = argparse.ArgumentParser(
        description="A universal document converter using pandoc.",
        formatter_class=argparse.RawTextHelpFormatter,
        epilog="""
Supported formats include:
- markdown (.md)
- latex (.tex)
- pdf (.pdf)
- html (.html)
- plain text (.txt)
        """
    )
    parser.add_argument("input", help="Input file to convert.")
    parser.add_argument("-o", "--output", required=True, help="Output file.")
    
    args = parser.parse_args()

    if not shutil.which("pandoc"):
        print("Error: pandoc is not installed or not in the system's PATH.", file=sys.stderr)
        sys.exit(1)

    pandoc_args = [
        "pandoc", args.input,
        "-o", args.output,
        "--pdf-engine=pdflatex",
        "-V", "fontsize=12pt",
        "-V", "documentclass=article",
        "-V", "geometry:margin=1in",
    ]

    try:
        print(f"Running command: {' '.join(pandoc_args)}")
        result = subprocess.run(pandoc_args, check=True, capture_output=True, text=True)
        print(f"Successfully converted '{args.input}' to '{args.output}'")
        if result.stdout:
            print(result.stdout)
    except subprocess.CalledProcessError as e:
        print(f"Error converting file:", file=sys.stderr)
        print(e.stderr, file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()
